### ISSUE TITLE
Support for raw data type in variables

### ISSUE THREAD

**Issue by dijitalmunky** [serverless/serverless#672]
It appears that serverless variables, while awesome, are limited to being strings.  While this works in most cases, there are a few where it does not.

Serverless should enhance it's variable replacement technology to be able to replace more than parts of strings.

**Comment by dijitalmunky** [comment]
The second case I came to this issue was wanting to put the `apiKeyRequired` variable for APIG configurations into an `s-variables-*.json` file.  So, based upon my experience in the previous comment, I placed this in `s-variables-common.json`:
```
{
  "apiKeyRequired": "false"
}
```
And then in my s-function.json file:
```
{
...
  "endpoints": [
    {
      "path": "api/{param}",
      "method": "GET",
      "type": "AWS",
      "apiKeyRequired": "${apiKeyRequired}"
    }
  ]
...
}
```

However upon deploy I received:
```
Serverless: Failed to deploy endpoints in "dev" to the following regions:
Serverless: us-east-1 ------------------------
Serverless:   GET - api/{param}: Expected params.apiKeyRequired to be a boolean
Serverless:
```
I also tried removing the quotes around the boolean value in `s-variables-*.json` but I then got the `b.replace is not a function` error as in my previous post.

**Comment by DanielRuskin1** [comment]
Looks like I'm still having trouble with this - could you post a sample with a boolean variable?

**Comment by eahefnawy** [comment]
@DanielRuskin1 while we now support any data type, not just strings, there's a bug with using boolean variables. There's an issue open for it, but I can't seem to find it :/ Once I do I'll let you know. Will push a fix for it asap.

Sorry for the inconvenience 😊

**Comment by DanielRuskin1** [comment]
Ah makes sense, I ended up using a workaround.  Thanks for the help!

**Comment by FreeWillaert** [comment]
Hi Daniel, we just bumped into the exact same issue, also trying to use a variable for apiKeyRequired - and we're not quite ready to move to 1.0 at this point.
Would you share your workaround with us?

**Comment by DanielRuskin1** [comment]
@FreeWillaert My workaround was specific to my use case unfortunately.  I initially wanted to only schedule a lambda in the dev environment (not prod), but ended up just scheduling it in both dev/prod.

Is this something that works for your use case?

**Comment by FreeWillaert** [comment]
Hi Daniel, we want to require an API key in tst/prd environment but not in dev - so your workaround won't help  unfortunately.
We found ticket https://github.com/serverless/serverless/issues/1744 which states that it will not be solved in v0.5 any more.

For the time being we'll replace the values in s-function.json manually or with gulp perhaps, and then move to v1.0 when we can.
